### DISCUSSION TITLE
gatsby-plugin-image: Missing image prop after upgrade to v4 (Netlify)

### DISCUSSION THREAD

**Discussion by MotaZor94** [gatsbyjs/gatsby#35631] [Help]
We are having an issue with displaying images on Netlify after upgrading Gatsby to V4.

We have been using Netlify for the past 3 years for this project and everything worked fine until we tried to move to V4 of Gatsby.

We are using Strapi with Gatsby and our source plugin is gatsby-source-graphql.

I am also using createRemoteFileNode API to download images and store them locally.

```
exports.createResolvers = async ({ actions, cache, createNodeId, createResolvers, store, reporter }) => {
    const { createNode } = actions;
    await createResolvers({
        STRAPI_UploadFile: {
            imageFile: {
                type: 'File',
                async resolve(source) {
                    let sourceUrl = `${source.url}`;
                    console.log('source Url:', sourceUrl);
                    return await createRemoteFileNode({
                        url: encodeURI(sourceUrl),
                        store,
                        cache,
                        createNode,
                        createNodeId,
                        reporter,
                    });
                },
            },
        },
    });
};
```
As you can see above in the gatsby-node.js I have tried to console.log the image urls and they appear in the netlify build process. Could it be that there is something not working as expected in createRemoteFileNode API?

Here is the list of the dependencies.
```
  "dependencies": {
    "@headlessui/react": "^1.5.0",
    "@heroicons/react": "^1.0.6",
    "@loadable/component": "^5.15.0",
    "babel-plugin-styled-components": "^1.12.0",
    "cross-env": "^7.0.3",
    "gatsby": "^4.13.1",
    "gatsby-plugin-google-tagmanager": "^4.0.0",
    "gatsby-plugin-image": "^2.13.0",
    "gatsby-plugin-manifest": "^4.13.0",
    "gatsby-plugin-offline": "4.0.0",
    "gatsby-plugin-postcss": "^4.0.0",
    "gatsby-plugin-react-helmet": "^5.13.0",
    "gatsby-plugin-sharp": "^4.13.0",
    "gatsby-source-filesystem": "^4.13.0",
    "gatsby-source-graphql": "^4.13.0",
    "gatsby-transformer-sharp": "^4.13.0",
    "iframe-resizer-react": "^1.1.0",
    "lodash": "^4.17.21",
    "moment-timezone": "^0.5.33",
    "prop-types": "15.7.2",
    "react": "17.0.1",
    "react-carousel": "^4.3.0",
    "react-dom": "17.0.1",
    "react-helmet": "6.1.0",
    "react-markdown": "^6.0.0",
    "react-modal": "^3.13.1",
    "react-spinners": "^0.11.0",
    "styled-components": "^5.3.0",
    "use-media": "^1.4.0"
  },
```
The images work when built locally. I have also tested this same setup on Gatsby Cloud and everything seemed to be working fine, but due to the business decisions, we are unable to migrate to Gatsby Cloud.

Did anyone experience anything similar to this, or do you have maybe a solution to what this would be?

Below is the screenshot of a small portion of our deploy log which is covered with warnings about images missing in the gatsby-plugin-image.

<img width="601" alt="Screen Shot 2022-05-11 at 10 14 57" src="https://user-images.githubusercontent.com/45624002/167812883-65e2e7dd-2fd3-4fee-aa42-ee452cdd4a89.png">

**Comment by LekoArts** [comment]
Hi!

This breaking change applies to you: https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v3-to-v4/#do-not-create-nodes-in-custom-resolvers

I'd highly recommend switching to https://github.com/strapi/gatsby-source-strapi for your sourcing.

**Reply by MotaZor94** [reply]
Hi Leko, thanks for the quick response.

So, what would be the best way to go through each remote image and generate a local file for it?

Unfortunately, we are still on Strapi V3 so we can't use that source plugin.

**Reply by LekoArts** [reply]
The `gatsby-source-graphql` doesn't create any nodes (smallest unit of data) in Gatsby's data store so you can't attach/create image nodes directly with it. We had to make this change to support SSR and DSG.

> If you’ve used this with gatsby-source-graphql, please switch to [Gatsby GraphQL Source Toolkit](https://github.com/gatsbyjs/gatsby-graphql-toolkit). Generally speaking you’ll want to create your own source plugin to fully support such use cases.

You'll need to create proper nodes through a source plugin, I think the easiest way would be using `gatsby-source-strapi` instead of writing your own one.

You might ask yourself: Well, why does it "work" in Gatsby Cloud or locally? It's kinda a race condition and it's not stable or guaranteed that it always succeeds. So you can't rely on this.

**Reply by MotaZor94** [reply]
Thanks for the answer @LekoArts
